### DISCUSSION TITLE
Integration causes Alarm to go offline - even in the normal phone App

### DISCUSSION THREAD

**Discussion by kimfr** [hawky358/hass_ids_hyyp#24] [General]
Hello,

Even since I installed this integration, I often have that my "Radio module" (GSM) goes offline so that both the HA integration and the normal ADT SecureHome App reports alarm offline. I had the same issue with the the RainerM26 version.

Speaking to ADT is completely useless as they just claim the GSM module is defect - yet after a few days, it normal comes back. If I however disable the integration, after sometime (latest at month roll-over) the alarm is back online - and stays that way.

It almost seems like to me that the SIM Card runs of out of data - yet I haven't been able to find anyone at ADT to confirm this.

I have been trying to check with wireshark (with integration enabled and without) - I get a difference of about 7 IP's - but it's not easy to figure out which ones are due to integration or which are simply due to DNS round-robin.

Does anyone have a similar issue?
Is there anyone else using the GSM module (ie. Radio module)?
How can I figure out how much traffic the integration generates against the ADT servers?

btw, I'm running HA Core container, version 2024.1.6 on a Raspberry Pi 4

**Comment by hawky358** [comment]
Hello,

Just some background:
- The normal HYYP module uses MQTT to communicate with the IDS server. 
- The app (and integration) uses http get requests to communicate with the IDS server.

- The HYYP module normally only "pushes" updates if there are any changes (e.g. arm, disarm, trigger, etc.), so very seldom if not using the integration.
- The HYYP module will also update to the server if polled (i.e. by refreshing the app, or by the HA integration every 30 seconds)

The amount of data that the module sends to the server isn't exactly the same as the amount that's received by the integration, but it's close enough to estimate usage.


- A 16 zone 2 partition setup generates about 6kB per request, but let's call it 10kB.
- A request under normal usage is every 30 seconds.
- So 86400/30 = 2880 requests a day @ 10kB a request ~ 29MB a day.  For 30 days a month so ~ 900MB a month.


I don't know anything about the GSM module. I am assuming it uses data connections and not SMS to communicate with the IDS server.
Perhaps IDS limits the amount of GSM requests per month or something.  The IP module (i.e. wifi / LAN) doesn't have this issue

**Comment by kimfr** [comment]
Thanks for the detailed information.

The strange thing is that I can conclusively tie my issues with the "offline GMS module" back to when I started using the integration - both your version back in October/November 2023 and previously in 2021/2022 when I was using the RainerM26 version. In 2021/2022, ADT even exchanged the GSM Module under warranty - same thing again though - worked for a few weeks, then stopped. Once I disabled the integration, no issue for 16 months until I activated your integration (because of the Push notification support).

btw, I have noticed that the iPhone App doesn't always do polling. If I turn off the Alarm on the Panel (for example) - I get the notification but if I then check the App several minutes later - it still says "Armed" - I have to "pull down" to refresh.

As you mentioned, I also suspect some short of "request" limit is reached - my suspicion would be the 30 second poll that Home Assistant does, is somehow causing a limit to be reached (what/where ever it is).

Is there anyway for me to check the communication between Home Assistant and the IDS/ADT servers?
(debug logs or something - I was hoping maybe there was a "too many requests error" somewhere ;-) )
Can I perhaps change the 30 second polling interval?
Is it perhaps "update_interval = timedelta(seconds=30)" in coordinator.py?
If so, do I just change that and restart HA or do I need to do something more?

My understanding is also that the GSM Module is using data - simply just pushing IP traffic over cellular.

**Comment by hawky358** [comment]
> btw, I have noticed that the iPhone App doesn't always do polling. If I turn off the Alarm on the Panel (for example) - I get the notification but if I then check the App several minutes later - it still says "Armed" - I have to "pull down" to refresh.

Correct, the app only polls when you "refresh". It normally also polls when opened from a completely closed state.
I am assuming the "ADT" app is very similar to the hyyp app.
The "ADT" version of the API also goes to the IDS servers, they just have a different packet header which I assume IDS somehow uses to charge ADT a license fee or something.

> Is there anyway for me to check the communication between Home Assistant and the IDS/ADT servers? (debug logs or something - I was hoping maybe there was a "too many requests error" somewhere ;-) ) 

The "simplest" way is for you to use the ([API](https://github.com/hawky358/pyHyypApi)) from a PC. You'll need python and PIP. 
The package is pyhyypapihawkmod so `pip install pyhyypapihawkmod==1.3.4` will get you the API. 
Sorry, I haven't really documented much on the API since it's mainly used with HA.  
If this is something you can do, let me know and I'll send you some snippits of code to get the data using the API.

I can also make a "debug" build of the integration again.  I typically remove all the debug info from the releases since I don't want the information to just sit there in a text file.  This may take some time, as I am a bit busy at the moment and I'll need to relook at the code that I haven't touched in 6 months.  Directly using the API is also a bit more flexible.


> Can I perhaps change the 30 second polling interval? Is it perhaps "update_interval = timedelta(seconds=30)" in coordinator.py? If so, do I just change that and restart HA or do I need to do something more?

Yes, that will change the polling interval. Change the value on your HA to e.g. timedelta(minutes=10). Then restart HA, now it will only poll every 10 minutes (or when you do something like arm/disarm/bypass - if you do too many things in rapid succession, you only get the 1st response since the server doesn't like to reply too frequently. You'll see this on the app as well.)

I did add functionality into the API where the system didn't use polling continuous polling, but rather when requested (i.e. refresh button / automation, or when arm/disarm etc.) similar to the app, but with an action buffer to wait for multiple actions.  I didn't really pursue it too much to implement into the integration since I hadn't implemented push notifications at that time yet. It would also break things such a HA notification if you've left a door open after 21:00 or something since you don't have a continuous check.


> My understanding is also that the GSM Module is using data - simply just pushing IP traffic over cellular.

That would make sense.  I have no reference though.

**Comment by kimfr** [comment]
> > Is there anyway for me to check the communication between Home Assistant and the IDS/ADT servers? (debug logs or something - I was hoping maybe there was a "too many requests error" somewhere ;-) )
> 
> The "simplest" way is for you to use the ([API](https://github.com/hawky358/pyHyypApi)) from a PC. You'll need python and PIP. The package is pyhyypapihawkmod so `pip install pyhyypapihawkmod==1.3.4` will get you the API. Sorry, I haven't really documented much on the API since it's mainly used with HA. If this is something you can do, let me know and I'll send you some snippits of code to get the data using the API.
> 
I'm afraid that I have zero pyhon knowledge... happy to try and snippits of code you can send.

> I can also make a "debug" build of the integration again. I typically remove all the debug info from the releases since I don't want the information to just sit there in a text file. This may take some time, as I am a bit busy at the moment and I'll need to relook at the code that I haven't touched in 6 months. Directly using the API is also a bit more flexible.
> 
Could be an option (see below)

> > Can I perhaps change the 30 second polling interval? Is it perhaps "update_interval = timedelta(seconds=30)" in coordinator.py? If so, do I just change that and restart HA or do I need to do something more?
> 
> Yes, that will change the polling interval. Change the value on your HA to e.g. timedelta(minutes=10). Then restart HA, now it will only poll every 10 minutes (or when you do something like arm/disarm/bypass - if you do too many things in rapid succession, you only get the 1st response since the server doesn't like to reply too frequently. You'll see this on the app as well.)
> 
I guess the easiest is to increase the polling and see what happens - I'll do that now and let's see how long the Alarm stays online - just need to it to come back online again first......

Will report back here once I have something.

**Comment by kimfr** [comment]
I spoke to the IDS HYYP technical hotline - the HYYP GSM Modules are allocated **50MB** of Cellular data per month.

I spoke to 2 different technicians - they were however not sure if a "App refresh" causes the HYYP servers to contact the GSM module - I guess it does as it would fit with what I have seen.

As mentioned, I have changed the update_interval to 10 minutes - let's see how that goes then.
Based on data usage assumption - this should work....

**Comment by hawky358** [comment]
> I spoke to the IDS HYYP technical hotline - the HYYP GSM Modules are allocated **50MB** of Cellular data per month.

ok, that's not going to work so well for this then

> I spoke to 2 different technicians - they were however not sure if a "App refresh" causes the HYYP servers to contact the GSM module - I guess it does as it would fit with what I have seen.

It 100% does. 
When "idle" (i.e. nothing triggered or arm/disarm etc.), the system sends no data.  The only time the server gets updated data is when something happens. When the app is refreshed the server requests updated data from the panel/module. So for a "typical house" you'd use maybe 5-10 refreshes a day.

I know this because I monitored the packets to and from the hyyp module. You can force the module into an unencrypted mqtt fallback mode and thus see the actual data being transmitted. 


> As mentioned, I have changed the update_interval to 10 minutes - let's see how that goes then. Based on data usage assumption - this should work....

Should help yes.  It's unfortunately a limitation of the hardware they provide.
I didn't even know about these hyyp GSM modules. I've got some ideas for a "low-data-mode" which will not poll at all unless you call a service from HA.  That way you can control the data you use and trigger it with a button or on notifications or something.
I'll see when I can get to writing and testing something, but I am away for the next week or so.

**Comment by kimfr** [comment]
> > As mentioned, I have changed the update_interval to 10 minutes - let's see how that goes then. Based on data usage assumption - this should work....
> 
> Should help yes. It's unfortunately a limitation of the hardware they provide. I didn't even know about these hyyp GSM modules. I've got some ideas for a "low-data-mode" which will not poll at all unless you call a service from HA. That way you can control the data you use and trigger it with a button or on notifications or something. I'll see when I can get to writing and testing something, but I am away for the next week or so.

As I'm most likely switching to OAlarm in October due to changing Armed Response Company - I can live with the current situation until then.

Happy to test if you decide to develop something.

**Comment by itbrl-gmail** [comment]
I have the same issue with HYYP. Thanks for the work with the integration.

**Reply by hawky358** [reply]
Do you also have the GSM module?

**Reply by itbrl-gmail** [reply]
Yes
